Library/TransactionError.cs

```diff
@@ -77,14 +83,17 @@ internal TransactionError(XmlTextReader reader)
                     case "gateway_error_code":
                         GatewayErrorCode = reader.ReadElementContentAsString();
                         break;
+                    case "three_d_secure_action_token_id":
+                        ThreeDSecureActionTokenId = reader.ReadElementContentAsString();
+                        break;
                 }
             }
         }
 
         public override string ToString()
         {
-            return string.Format("Code: \"{0}\" Category: \"{1}\" CustomerMessage: \"{2}\" MerchantAdvice: \"{3}\" GatewayCode: \"{4}\""
-                , ErrorCode, ErrorCategory, CustomerMessage, MerchantAdvice, GatewayErrorCode);
+            return String.Concat(this.GetType().GetProperties()
+                .Select(i=>string.Format("{0}: \"{1}\"", i.Name, i.GetValue(this, null))));
```

Comment:
This was getting out of hand if I were to add another property to print out. I figured, why not just print all the values that we need rather than printing them one by one?

If we need to clean up formatting on this at all let me know.